### PR TITLE
Sharanya test

### PR DESCRIPTION
**Pull request by sahilaug**

### CODE REVIEW

**Comment by sharanya5** [comment]
@sahilaug please review the changes added... thank you...

File: src/App.js [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
-                                 {item : { id : 4,
-                                          name : "p4",
-                                          age : 20} } ];
-       return ( <div>
-                  <ContactListPage data={ contacts } /> 
-                </div>);
-      }
+    
+    render() {
+        const contacts= fromJS([ { id : 1,
```
Comment (sahilaug) [inline src/App.js]: You can initialise it in ContactListPage.js and render only the ContactListPage here in App.js

File: src/ContactListPage.js [hunk 2]
```diff
@@ -1,26 +1,21 @@
...
-                  <button type="button">add new</button>
-                </div>
-              </div>);
+  render() {
+
+       return ( <div>
+                  <div>
+                    <h1 className='App'>contact list</h1>
+                      </div> 
+                        <ContactList toItems={ this.props.data } />
```
Comment (sahilaug) [inline src/ContactListPage.js]: Rename prop 'toItems' to a more meaningful name something like 'contactList' = { contactList }. Makes more sense this way.

File: src/ContactList.js [hunk 3]
```diff
@@ -1,21 +1,96 @@
...
 import ContactCard from './ContactCard';
+import { fromJS, toJS } from 'immutable';
 
 class ContactList extends Component {
-    constructor(props){
+    
+    constructor(props) {
         super(props);
-        this.state={contactItems: props.toItems }
+            this.state = { contactItems: props.toItems, newItem: {} , newVar : false }
```
Comment (sahilaug) [inline src/ContactList.js]: Rename 'newVar' to something more meaningful eg 'showAddNewContactFlag': false

File: src/ContactList.js [hunk 4]
```diff
@@ -1,21 +1,96 @@
...
+            }
+            else return item;
+        });
+        // console.log({ newArray: newArray.toJS() })
+        console.log(this.state.contactItems.toJS())
+        this.setState({ contactItems : newArray })
+        console.log(this.state.contactItems.toJS())
+    }
+
+    newContactCards(e, field) {
```
Comment (sahilaug) [inline src/ContactList.js]: Rename this method

File: src/ContactList.js [hunk 5]
```diff
@@ -1,21 +1,96 @@
...
+        this.state.newItem[field] = e.target.value;
+    }
+
+    addContactCard() {
+        this.setState({ contactItems: this.state.contactItems.push(fromJS(this.state.newItem)) });
+    }
+
+    contactCardForm(){
```
Comment (sahilaug) [inline src/ContactList.js]: Rename to something which conveys the meaning of the method

File: src/ContactList.js [hunk 6]
```diff
@@ -1,21 +1,96 @@
...
+                <h2 className="header">New Contact </h2>
+                    <div className="secdiv">
+                        New id   : <input type="text" Name="id" onChange={(e)=>this.newContactCards(e,'id')}/> <br/>
+                        New Name : <input type="text" Name="name" onChange={(e)=>this.newContactCards(e,'name')}/><br/>
+                        New Age  : <input type="text" Name="Age" onChange={(e)=>this.newContactCards(e,'age')} /><br/>
+                        New num : <input type="text" Name="ph_no" onChange={(e)=>this.newContactCards(e,'ph_no')}/>
+                    </div>
+                    <div>
+                        <button type="button" className="addButton" onClick={()=>this.addContactCard()}>Add</button>
+                        <button type="button" className="resetButton"><a href="/">Reset</a></button>
```
Comment (sahilaug) [inline src/ContactList.js]: Add the method which resets the state of add new contact form

File: src/ContactCard.js [hunk 7]
```diff
@@ -1,24 +1,56 @@
...
-        this.state.card.pop(this.props.id);
+    
+constructor(props) {
+    super(props);
+    this.state={ cardItem: props.contactData, editMode: false }
+    this.newCard=this.newCard.bind(this);
+    this.formDisplay=this.formDisplay.bind(this);
+}
+
+newCard(e, field)  {
```
Comment (sahilaug) [inline src/ContactCard.js]: better name

File: src/ContactCard.js [hunk 8]
```diff
@@ -1,24 +1,56 @@
...
+    var newValue = e.target.value; 
+    this.setState({cardItem : newValue});
+    this.props.editCard(this.state.cardItem.id, newValue, field);
+}
+
+componentWillReceiveProps(nextProps) {
+    this.setState({ cardItem: nextProps.contactData })
+}
+
+formDisplay(e, id){
```
Comment (sahilaug) [inline src/ContactCard.js]: better name

File: src/ContactCard.js [hunk 9]
```diff
@@ -1,24 +1,56 @@
...
+    this.setState({editMode : !this.state.editMode})
+}
+
+render() {
+    var cls = 'foredit' + (this.state.editMode ? 'visible' : '')
```
Comment (sahilaug) [inline src/ContactCard.js]: Why two different kinds of logics to showHide forms in ContactCard.js and ContactList.js. Try to make both of them uniform inside the same project.

File: src/ContactCard.js [hunk 10]
```diff
@@ -1,24 +1,56 @@
...
+    return(
+        <div className='Cards' >
+                <ol draggable='true'>
```
Comment (sahilaug) [inline src/ContactCard.js]: draggable='true'. is it required ?

**Review by sahilaug — COMMENTED** [review]
@sharanya5 Please take care of resolving review comments

File: src/redux/ContactList.js [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+    }),
+    deleteContact: (payload) => ({
+        type: constants.DELETE_CONTACT,
+        payload
+    }),
+    editContact: (payload) => ({
+        type: constants.EDIT_CONTACT,
+        payload
+    }),
+    addNewContact: (payload) => ({
```
Comment (sahilaug) [inline src/redux/ContactList.js]: AddNewContact and AddContact to list should be the same

File: src/ContactList.js [hunk 12]
```diff
@@ -1,21 +1,105 @@
...
+    editCard(id, newValue, field) {
+        this.props.editContact({id,newValue,field});
+    }
+
+    addNewContactCard(e, field) {
+        // this.props.newContact({ item: { [field]: e.target.value } });
+        const item = {};
+        item[field] = e.target.value;
+        console.log(item);
+        this.props.addNewContact( {item} )
```
Comment (sahilaug) [inline src/ContactList.js]: I am not sure why are you calling addNewContact action in the onChange handler?

Comment (sahilaug) [reply src/ContactList.js]: Just set the react state for newItem, snd while submitting the add button, call the AddCardtolist action with the this.state.newItem as payload

Comment (sharanya5) [reply src/ContactList.js]: instead of 2 functions i should make it as one function.. and it should be called on button click???

Comment (sahilaug) [reply src/ContactList.js]: onChange handler should only set the newItem in the react state, where as onSubmit for the add  button should call the action with newItem in the payload

**Pull request merged**